src/navigation/wallet/screens/wallet-settings/ExportTransactionHistory.tsx

```diff
@@ -0,0 +1,291 @@
+        ),
+      );
+
+      // @ts-ignore
+      const {unitToSatoshi} = dispatch(
+        GetPrecision(currencyAbbreviation, chain),
+      );
+      const satToUnit = 1 / unitToSatoshi;
+
+      let _amount, _note, _copayers, _creator, _comment;
+      const csvContent: any[] = [];
+
+      transactions.forEach(tx => {
+        let amount = tx.amount;
+
+        if (tx.action == 'moved') {
+          amount = 0;
+        }
+
+        _copayers = '';
+        _creator = '';
+
+        if (tx.actions && tx.actions.length > 1) {
+          for (let i = 0; i < tx.actions.length; i++) {
+            _copayers +=
+              tx.actions[i].copayerName + ':' + tx.actions[i].type + ' - ';
+          }
+          _creator = tx.creatorName ? tx.creatorName : '';
+        }
+        _amount =
+          (tx.action == 'sent' ? '-' : '') + (amount * satToUnit).toFixed(8);
+        _note = tx.message || '';
+        _comment = tx.note ? tx.note.body : '';
+
+        if (tx.action == 'moved') {
+          _note += ' Sent to self:' + (tx.amount * satToUnit).toFixed(8);
+        }
+
+        csvContent.push({
+          Date: formatDate(tx.time * 1000),
+          Destination: tx.addressTo || '',
+          Description: _note,
+          Amount: _amount,
+          Currency: currencyAbbreviation.toUpperCase(),
+          Txid: tx.txid,
+          Creator: _creator,
+          Copayers: _copayers,
+          Comment: _comment,
+        });
+
+        if (tx.fees && (tx.action == 'moved' || tx.action == 'sent')) {
+          const _fee = (tx.fees * satToUnit).toFixed(8);
+          csvContent.push({
+            Date: formatDate(tx.time * 1000),
+            Destination: `${chain.toUpperCase()} Network Fees`,
+            Description: _note,
+            Amount: '-' + _fee,
+            Currency: chain.toUpperCase(),
+            Txid: '',
+            Creator: '',
+            Copayers: '',
+          });
+        }
+      });
+      const csv = Papa.unparse(csvContent);
+      return csv;
+    } catch (e) {
+      const errString = e instanceof Error ? e.message : JSON.stringify(e);
+      dispatch(LogActions.warn(`[buildCVSFile]: ${errString}`));
+      throw e;
+    }
+  };
+
+  const shareFile = async (csv: any, option: string) => {
+    try {
+      if (Platform.OS === 'android') {
+        await isAndroidStoragePermissionGranted();
+      }
+      const rootPath =
+        Platform.OS === 'ios'
+          ? RNFS.LibraryDirectoryPath
+          : RNFS.TemporaryDirectoryPath;
+      const csvFilename = `${APP_NAME_UPPERCASE}-${walletName}`;
+      let filePath = `${rootPath}/${csvFilename}`;
+
+      await RNFS.mkdir(filePath);
+
+      filePath += '.csv';
+      const opts = {
+        title: csvFilename,
+        url: `file://${filePath}`,
+        subject: `${walletName} Transaction History`,
+      };
+
+      await RNFS.writeFile(filePath, csv, 'utf8');
+
+      if (option === 'download') {
+        await Share.open(opts);
+      } else {
```

Comment:
I'd change this part with a native mail plugin `react-native-mail` in order to check if the user has defined default email client. Also we could hide if it's not available.

Reply:
nice! I'll make the changes, ty!